modules/apps/message-boards/message-boards-web/src/main/resources/META-INF/resources/message_boards/js/MBPortlet.es.js

```diff
@@ -12,225 +12,243 @@
  * details.
  */
 
-import {PortletBase, fetch} from 'frontend-js-web';
-import core from 'metal';
-import {EventHandler} from 'metal-events';
+import {fetch} from 'frontend-js-web';
 
 const RECENTLY_REMOVED_ATTACHMENTS = {
 	multiple: Liferay.Language.get('x-recently-removed-attachments'),
 	single: Liferay.Language.get('x-recently-removed-attachment'),
 };
 
+const CONFIRM_DISCARD_IMAGES = Liferay.Language.get(
+	'uploads-are-in-progress-confirmation'
+);
+
 /**
  * MBPortlet handles the actions of replying or editing a
  * message board.
- *
- * @abstract
- * @extends {PortletBase}
  */
 
-class MBPortlet extends PortletBase {
-
-	/**
-	 * @inheritDoc
-	 */
-
-	created() {
-		this.eventHandler_ = new EventHandler();
+export default function MBPortlet({
+	constants,
+	currentAction,
+	getAttachmentsURL,
+	namespace,
+	replyToMessageId,
+	rootNodeId,
+	strings = {
+		confirmDiscardImages: CONFIRM_DISCARD_IMAGES,
+	},
+	viewTrashAttachmentsURL,
+}) {
+	const rootNode = document.getElementById(rootNodeId);
+
+	const workflowActionInputNode = rootNode.querySelector(
+		`#${namespace}workflowAction`
```

Comment:
Not a big deal, but it feels weird to do a scoped `querySelector` by `id`.

There should be just one node in the whole document with id `#${namespace}workflowAction`, so this should be equivalent to just querying the whole document. Not that I think makes a lot of difference, but it could be shadowing errors if that is not the case.